contracts/OctobayGovNFT.sol

```diff
@@ -46,6 +46,16 @@ contract OctobayGovNFT is OctobayStorage, ERC721Pausable {
         _grantPermission(_tokenId, Permission.CREATE_PROPOSAL);
     }
 
+    /// @notice Revokes all permissions from the given NFT
+    /// @param _tokenId ID of the NFT to grant permissions to
+    function revokeAllPermissions(uint256 _tokenId) internal {
+        // There's no nice way of looping through enums... :( It's probably better that we do this here though
+        _revokePermission(_tokenId, Permission.MINT);
+        _revokePermission(_tokenId, Permission.TRANSFER);
+        _revokePermission(_tokenId, Permission.SET_ISSUE_GOVTOKEN);
+        _revokePermission(_tokenId, Permission.CREATE_PROPOSAL);
+    }    
```

Comment:
Isn't that more or less the same as burning? Well, of course it's not. You can re-grant permissions. But you know what I mean? The function I envision here has a different name and it's called "suspend" and only changes a single boolean flag. What do you think?

Ok, just saw it actually IS used in the burn function. But since this is probably the only use case for that function, I think the code can be directly in burn function. Again... what do you think? :smiley:

Reply:
So the other thing I was thinking was - do we need to revoke at all? I have it in there as a precaution at the moment, but if we burn the token, the reference we keep for its permissions isn't relevant anymore. Might there be cases when someone looks directly at the permissions mapping from a cached tokenId?

If we do want it though, yeah I can't really see a use besides when burning so we could move it there.

Reply:
> Might there be cases when someone looks directly at the permissions mapping from a cached tokenId?

Can't think of any situation where that might happen. We index stuff in the subgraph but also for burning/deleting stuff. And what should be the problem with it anyway? Once the token is burned... its burned and old permissions are indeed irrelevant.

Reply:
Right, there's an event for burn as well so there really should be no reason to look at token permissions for something that's burnt. I'll remove it then.